app/components/quick_text_input.js

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {TextInput} from 'react-native';
+
+// A component that can be used to make partially-controlled inputs that can be updated
+// by changing the value prop without lagging the UI.
+//
+// We're using this in place of a connected TextInput due to changes made in RN v0.54
+// that break input in Chinese and Japanese when using a connected TextInput. See
+// https://github.com/facebook/react-native/issues/18403 for more information.
+//
+// In addition to that, there's also an ugly hack to change the key on the TextInput
+// when this is triggered because that same version made setNativeProps work inconsistently.
+// See https://github.com/facebook/react-native/issues/18272 for more information on that.
+export default class QuickTextInput extends React.PureComponent {
+    static propTypes = {
+
+        onChangeText: PropTypes.func,
+
+        /**
+         * The string value displayed in this input
+         */
+        value: PropTypes.string.isRequired,
+    };
+
+    static defaultProps = {
+        delayInputUpdate: false,
+        value: '',
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.storedValue = props.value;
+
+        this.state = {
+            key: 0,
+        };
+    }
+
+    componentDidMount() {
+        this.updateInputFromProps();
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
```

Comment:
why not use `getDerivedStateFromProps` ??

Reply:
getDerivedStateFromProps is static, but we need to access `this.storedValue`. Similarly, I would've used getSnapshotBeforeUpdate, but you can't mix the old and new methods